package.json

```diff
@@ -37,12 +37,14 @@
     "chokidar": "^3.3.0",
     "codename": "^0.0.6",
     "common-tags": "^1.8.0",
+    "content-type": "^1.0.4",
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
-    "express-graphql": "^0.9.0",
+    "fp-ts": "^2.5.4",
```

Comment:
I'm not convinced by the need of something as fancy as `fp-ts` yet. As a first step, we could've just created our own internal either type, or either-like type. While I see great patterns that we could achieve with `fp-ts`, we need to make sure that its types don't ever land on the public API, because even something as popular as the Either type in the fp world is not common at all in the JS world

Reply:
Yeah we’ll definitely not want the FP API/types escaping the implementation. It might mean we need to eventually maintain a thin layer on top. Really whatever helps us the most... the type safety added by not having untyped throw is great for us internally for example. Add more and more benefits like this and I _could_ see reaching a point where net velocity/simplicity for us is had by adopting the fp implementation with a thin non-FP layer. 

Internally I think it’s worth using the gold standard lib to properly appraise the techniques. It’s not like we’re adding many mb or kb to the nexus package. And the path forward to more FP features that rigorously compose is available when we’re ready. Like rx and lodash it is free shakable so once we care we can look into that.